### PR TITLE
Implement UI for editing a tag

### LINKED ISSUE #160: Implement UI for Edit a tag
- Edit the Tag
- Show Details
- Delete the Tag (with Overlay)

### PR DESCRIPTION
**Pull request by JanEberlein**
closes #160 and #323

**You will have to change the nfc-tags.db: either delete and add new tags or change values in spotifyTrackUri --> Remove spotify:track: from the value**

You could break the editing action at the moment if you try to edit a tag that doesn't have data (Just klick at continue in the add process) This was ignored because we will catch this issue in the AddTag-Flow (discussed with @anbraten)

### CODE REVIEW

File: packages/app/src/views/TagDetails.vue [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      class="h-2/3 overflow-x-hidden overflow-y-scroll bg-gradient-to-b from-primary to-secondary flex items-center justify-center text-4xl text-gray-800"
+    ></main>
+    <footer class="flex-grow flex items-center justify-center text-4xl text-gray-800">
+      <Button :text="'Speichern'" />
+    </footer>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref } from 'vue';
```
Comment (julianhahn) [inline packages/app/src/views/TagDetails.vue:16]: `
```suggestion
import { defineComponent } from 'vue';
```

unused ref

File: packages/app/src/router/index.ts [hunk 2]
```diff
@@ -16,6 +16,12 @@ const routes: RouteRecordRaw[] = [
     component: () => import('../views/Welcome.vue'),
     meta: { authentication: 'guests-only' },
   },
+  {
+    path: '/tag-details/:id',
```
Comment (julianhahn) [inline packages/app/src/router/index.ts:20]:
```suggestion
    path: '/tag/:id',
```

I would remove hypfen from the uri.

**Review by julianhahn — CHANGES_REQUESTED** [review]

File: packages/app/src/views/TagDetails.vue (hunk 1 above)
Comment (julianhahn) [reply packages/app/src/views/TagDetails.vue:16]: Sorry, didn't see this was on draft

File: packages/app/src/router/index.ts (hunk 2 above)
Comment (JanEberlein) [reply packages/app/src/router/index.ts:20]: good idea

**Review by tcoro — CHANGES_REQUESTED** [review]
good work, just minor UI changes:

image

why is there white whitespace left and right?
and could you hide the scrollbar?
edit: also the footer and header have different heights compared to other views

**Comment by JanEberlein** [comment]
> good work, just minor UI changes:
> 
> why is there white whitespace left and right?
> and could you hide the scrollbar?
> edit: also the footer and header have different heights compared to other views
> edit2: I would put the change image button next to the image, not below it, better us of the whitespace. Or make the image the whole width

working on it

**Comment by JanEberlein** [comment]
> > good work, just minor UI changes:
> > why is there white whitespace left and right?
> > and could you hide the scrollbar?
> > edit: also the footer and header have different heights compared to other views
> > edit2: I would put the change image button next to the image, not below it, better us of the whitespace. Or make the image the whole width
> 
> working on it

done (except for uniform header and footer sizes, I opened issue #325 for that)

**Review by julianhahn — APPROVED** [review]
looks good, now with working checks for empty fields

**Review by tcoro — APPROVED** [review]
looks good

**Pull request merged**